Explorer/Assets/DCL/Backpack/AvatarSection/Outfits/Services/AvatarScreenshotService.cs

```diff
@@ -0,0 +1,185 @@
+        {
+            this.selfProfile = selfProfile;
+            if (!Directory.Exists(baseOutfitsDirectory))
+                Directory.CreateDirectory(baseOutfitsDirectory);
+        }
+
+        public async UniTask<Texture2D> TakeAndSaveScreenshotAsync(CharacterPreviewControllerBase controller, int slotIndex, CancellationToken ct)
+        {
+            string? userId = await GetCurrentUserIdAsync(ct);
+            if (string.IsNullOrEmpty(userId))
+            {
+                ReportHub.LogError(ReportCategory.OUTFITS, "Cannot save screenshot, user ID is not available.");
+                return null;
+            }
+
+            try
+            {
+                await UniTask.SwitchToMainThread(ct);
+
+                controller.SetPlatformVisible(false);
+                controller.ResetAvatarMovement();
+                controller.ResetZoom();
+                await UniTask.DelayFrame(2, PlayerLoopTiming.LastPostLateUpdate, ct);
+                await UniTask.Yield(PlayerLoopTiming.Update, ct);
+
+                var source = controller.CurrentRenderTexture;
+                if (source == null) return null;
+
+                ReportHub.Log(ReportCategory.OUTFITS, $"[Screenshot] Source RenderTexture sRGB: {source.sRGB}");
+                ReportHub.Log(ReportCategory.OUTFITS, $"[Screenshot] Source RenderTexture format: {source.graphicsFormat}");
+                int targetWidth = source.width / 2;
+                int targetHeight = source.height / 2;
+
+                // Get a temporary, smaller RenderTexture. This is highly optimized.
+                var tempRT = RenderTexture.GetTemporary(targetWidth, targetHeight, 0, source.format);
+
+                // Use Graphics.Blit to perform a GPU-accelerated copy and scale operation.
+                // This is the fastest way to downscale a texture.
+                Graphics.Blit(source, tempRT);
+
+                // Create the final Texture2D with the correct small dimensions.
+                var screenshotTexture = new Texture2D(targetWidth, targetHeight, TextureFormat.RGBA32, false);
+
+                // Set the temporary RT as active and read its (now small) contents.
+                RenderTexture.active = tempRT;
+                screenshotTexture.ReadPixels(new Rect(0, 0, targetWidth, targetHeight), 0, 0);
+
+                var pixels = screenshotTexture.GetPixels();
+                for (int p = 0; p < pixels.Length; p++)
+                {
+                    pixels[p] = pixels[p].gamma;
+                }
+
+                screenshotTexture.SetPixels(pixels);
+                
+                
+                screenshotTexture.Apply();
+
+                // Clean up: release the temporary RT and reset the active one.
+                RenderTexture.active = null;
+                RenderTexture.ReleaseTemporary(tempRT);
+
+                ct.ThrowIfCancellationRequested();
+
+                byte[] pngBytes = screenshotTexture.EncodeToPNG();
```

Comment:
I believe you can even avoid manipulating pixels in a managed form: you can get a pixel native array and pass it to https://docs.unity3d.com/ScriptReference/ImageConversion.EncodeNativeArrayToPNG.html